### PR TITLE
New Location Page Header

### PR DESCRIPTION
**Pull request by chasulin**
Implements the new location page header! New additions with this header are:
- A claim of 'government verified data' for applicable states (currently only NV)
- A risk level thermometer

Still to do:
- [x] Adjust the thermometer to include the new 4th color once Pablo's PR is merged in
- [x] Implement any feedback that will come from Josh

Screen Shot 2020-06-01 at 7 20 51 PM

Screen Shot 2020-06-01 at 7 21 10 PM

### CODE REVIEW

File: src/components/LocationPage/ChartsHolder.tsx [hunk 1]
```diff
@@ -154,13 +155,16 @@ const ChartsHolder = (props: {
...
-            <LocationPageHeader projections={props.projections} />
-            <SummaryStats
+            <NewLocationPageHeader
+              projections={props.projections}
               stats={getChartSummarys(projection)}
               onRtRangeClick={() => scrollTo(rtRangeRef.current)}
               onTestPositiveClick={() => scrollTo(testPositiveRef.current)}
               onIcuUtilizationClick={() => scrollTo(icuUtilizationRef.current)}
               onContactTracingClick={() => scrollTo(contactTracingRef.current)}
+              onHeaderShareClick={() => scrollTo(shareBlockRef.current, -215)}
```
Comment (goldblatt) [inline src/components/LocationPage/ChartsHolder.tsx:165]: llol

File: src/components/LocationPage/NewLocationPageHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  onHeaderSignupClick: () => void;
+  isMobile?: Boolean;
+}) => {
+  const hasStats = !!Object.values(props.stats).filter(
+    (val: number | null) => !isNull(val),
+  ).length;
+
+  const headerTopMargin = !hasStats ? -268 : -330;
+  const headerBottomMargin = !hasStats ? 115 : 0;
+
```
Comment (goldblatt) [inline src/components/LocationPage/NewLocationPageHeader.tsx:90]: why are these so ... specifc?

File: src/components/LocationPage/NewLocationPageHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  const locationName =
```
Comment (goldblatt) [inline src/components/LocationPage/NewLocationPageHeader.tsx:91]: does projections not have a location name ?

File: src/components/LocationPage/ShareButtons.style.js [hunk 4]
```diff
@@ -47,7 +47,7 @@ export const SaveOrShareButton = styled.div`
   ${StyledShareButtonStyles};
   border-right: ${({ isLast }) =>
     !isLast && `1px solid ${COLOR_MAP.GRAY.LIGHT}`};
-  color: #3b94e6;
+  color: #00bfea;
```
Comment (goldblatt) [inline src/components/LocationPage/ShareButtons.style.js:50]: should these come from the color file?

File: src/components/LocationPage/ShareButtons.style.js [hunk 5]
```diff
@@ -56,6 +56,7 @@ export const SaveOrShareButton = styled.div`
   justify-content: center;
   align-items: center;
   text-decoration: none;
+  front-weight: 500;
```
Comment (goldblatt) [inline src/components/LocationPage/ShareButtons.style.js:59]: font**?

Comment (chasulin) [reply src/components/LocationPage/ShareButtons.style.js:59]: oh.

File: src/components/LocationPage/NewLocationPageHeader.tsx (hunk 3 above)
Comment (chasulin) [reply src/components/LocationPage/NewLocationPageHeader.tsx:91]:
```
  const locationName =
    props.projections.countyName || props.projections.stateName;
```

File: src/components/LocationPage/NewLocationPageHeader.tsx (hunk 2 above)
Comment (chasulin) [reply src/components/LocationPage/NewLocationPageHeader.tsx:90]: Hmm yeah, so the header is drawn up onto the colored banner, and the margins change imprecisely depending on if the header includes stats or is just text.

Screen Shot 2020-06-05 at 7 15 38 PM

Screen Shot 2020-06-05 at 7 15 54 PM

File: src/components/LocationPage/ChartsHolder.tsx [hunk 6]
```diff
@@ -58,13 +58,13 @@ type County = {
   population: string;
 };
 
-const scrollTo = (div: null | HTMLDivElement) =>
+const scrollTo = (div: null | HTMLDivElement, offset: number = 180) =>
   div &&
   window.scrollTo({
     left: 0,
     // TODO: 180 is rough accounting for the navbar and searchbar;
     // could make these constants so we don't have to manually update
```
Comment (mikelehen) [inline src/components/LocationPage/ChartsHolder.tsx:66]: This todo should probably move up by the offset param now.

File: src/components/LocationPage/NewLocationPageHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+          stats={props.stats}
+          onRtRangeClick={props.onRtRangeClick}
+          onTestPositiveClick={props.onTestPositiveClick}
+          onIcuUtilizationClick={props.onIcuUtilizationClick}
+          onContactTracingClick={props.onContactTracingClick}
+          isMobile={props.isMobile}
+        />
+        <HeaderSection>
+          <HeaderSubCopyWrapper isVerifiedState={isVerifiedState}>
+            {props.projections.isCounty && !isEmbed && (
```
Comment (mikelehen) [inline src/components/LocationPage/NewLocationPageHeader.tsx:151]: Seems like all of this embed-specific stuff could be removed once this is merged with embeds 2.0?  (not urgent)

File: src/components/LocationPage/NewLocationPageHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+  ) : (
+    <strong>{props.projections.stateName}</strong>
+  );
+
+  return <span>{displayName}</span>;
+};
+
+const noop = () => {};
+
+const NewLocationPageHeader = (props: {
```
Comment (mikelehen) [inline src/components/LocationPage/NewLocationPageHeader.tsx:72]: What's the rationale for having NewLocationPageHeader.tsx and still keeping LocationPageHeader.js ?  Just temporary?  Should we have a TODO to delete LocationPageHeader.js and rename NewLocationPageHeader?

Comment (chasulin) [reply src/components/LocationPage/NewLocationPageHeader.tsx:73]: Yep! Just temporary. I plan on removing the old header+old embed after everything is deployed smoothly. I added in a TODO

File: src/components/LocationPage/NewLocationPageHeader.tsx (hunk 7 above)
Comment (chasulin) [reply src/components/LocationPage/NewLocationPageHeader.tsx:152]: Yeah, same as below- I'm gonna rip out the old embed+old header code once the new ones are deployed

**Comment by chasulin** [comment]
Oooh yes, fixed the button! Good catch, thanks!

**Pull request merged**